src/test/ui/unsafe/issue-85435-unsafe-op-in-let-under-unsafe-under-closure.rs

```diff
@@ -0,0 +1,25 @@
+// check-pass
```

Comment:
Should this test the THIR unsafeck as well, maybe?
```suggestion
// check-pass
// revisions: mir thir
// [thir]compile-flags: -Z thir-unsafeck
```
I'm not sure it's particularely useful for this issue though

Reply:
@LeSeulArtichaut that's a good idea, since it seems like our existing test suite is not exercising this code path well enough.

Reply:
(of course you are pointing out that the *fix* added in this PR has nothing to do with THIR. but still, better to make the test happen. Though perhaps I should have waited to put it into a followup PR, I admit.)